### PR TITLE
Bundle-Adjustment-in-the-Large Example

### PR DESCRIPTION
**Pull request by aaron-skydio**

### CODE REVIEW

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+#include "./gen/keys.h"
+#include "./gen/snavely_reprojection_factor.h"
+
+using namespace sym::Keys;
+
+sym::Factord MakeFactor(int camera, int point, int pixel) {
+  return sym::Factord::Hessian(sym::SnavelyReprojectionFactor<double>,
+                               {
+                                   sym::Key::WithSuper(CAM_T_WORLD, camera),
```
Comment (hayk-skydio) [inline symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:22]: Curious if there's a reason this is with a superscript vs subscript via `sym::Key(CAM_T_WORLD, camera)` or `{CAM_T_WORLD, camera}`

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                                   sym::Key::WithSuper(INTRINSICS, camera),
+                                   sym::Key::WithSuper(POINT, point),
+                                   sym::Key::WithSuper(PIXEL, pixel),
+                                   EPSILON,
+                               },
+                               {
```
Comment (hayk-skydio) [inline symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:28]: maybe we add a prefix like `/* optimized_keys */` for clarity

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+struct Problem {
+  std::vector<sym::Factord> factors;
+  sym::Valuesd values;
+  int num_cameras;
+  int num_points;
+  int num_observations;
+};
+
+/// Read the problem description from the given path
```
Comment (hayk-skydio) [inline symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:43]: use our docstring format

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  sym::internal::SetLogLevel("info");
+
+  SYM_ASSERT(argc == 2);
+  const auto problem = ReadProblem(argv[1]);
+
+  sym::Valuesd optimized_values = problem.values;
+  sym::Optimizerd optimizer{sym::DefaultOptimizerParams(), std::move(problem.factors)};
+  const auto stats = optimizer.Optimize(&optimized_values);
+  spdlog::info("Finished in {} iterations", stats.iterations.size());
+}
```
Comment (hayk-skydio) [inline symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:112]: Very clean!

File: test/symforce_examples_bundle_adjustment_in_the_large_codegen_test.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+from pathlib import Path
+
+from symforce.examples.bundle_adjustment_in_the_large.bundle_adjustment_in_the_large import generate
+from symforce.test_util import TestCase, sympy_only
+
+SYMFORCE_DIR = Path(__file__).parent.parent
+
+
+class BundleAdjustmentInTheLargeCodegenTest(TestCase):
```
Comment (hayk-skydio) [inline test/symforce_examples_bundle_adjustment_in_the_large_codegen_test.py:14]: docstring

**Review by hayk-skydio — APPROVED** [review]
Very nice!

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc (hunk 1 above)
Comment (aaron-skydio) [reply symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:22]: `values_codegen.generate_values_keys` will generate keys with subscripts, which is necessary if you have more than 52 keys or so.  In this case all of our generated keys happen to not have subscripts, but the general way to use this mechanism (at least the way it's currently designed) is for the subscripts to be reserved for codegen time and superscripts for runtime

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc (hunk 2 above)
Comment (aaron-skydio) [reply symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:28]: Done

File: symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc (hunk 3 above)
Comment (aaron-skydio) [reply symforce/examples/bundle_adjustment_in_the_large/bundle_adjustment_in_the_large.cc:43]: Done

File: test/symforce_examples_bundle_adjustment_in_the_large_codegen_test.py (hunk 5 above)
Comment (aaron-skydio) [reply test/symforce_examples_bundle_adjustment_in_the_large_codegen_test.py:14]: Done

**Comment by aaron-skydio** [comment]
Closed by bad1035

**Pull request closed without merging**